packages/catalog-model/src/entity/policies/ForeignRootFieldsEntityPolicy.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EntityPolicy } from '../../types';
+import { Entity } from '../Entity';
+
+const defaultKnownFields = ['apiVersion', 'kind', 'metadata', 'spec'];
+
+/**
+ * Ensures that there are no foreign root fields in the entity.
+ */
+export class ForeignRootFieldsEntityPolicy implements EntityPolicy {
```

Comment:
Probably call this `NoForeignRootFieldsEntityPolicy`?

Reply:
definitely, fixed